lib/Raven/ClientInterface.php

```diff
@@ -0,0 +1,234 @@
+<?php
+
+/*
+ * This file is part of Raven.
+ *
+ * (c) Sentry Team
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Raven;
+
+use Raven\Breadcrumbs\Breadcrumb;
+use Raven\Context\Context;
+use Raven\Context\RuntimeContext;
+use Raven\Context\ServerOsContext;
+use Raven\Context\TagsContext;
+use Raven\Processor\ProcessorInterface;
+
+/**
+ * This interface must be implemented by all Raven client classes.
+ *
+ * @author Stefano Arlandini <sarlandini@alice.it>
+ */
+interface ClientInterface
+{
+    /**
+     * Gets the configuration of the client.
+     *
+     * @return Configuration
+     */
+    public function getConfig();
+
+    /**
+     * Gets the transaction stack.
+     *
+     * @return TransactionStack
+     */
+    public function getTransactionStack();
+
+    /**
+     * Adds a new middleware with the given priority to the stack.
+     *
+     * @param callable $middleware The middleware instance
+     * @param int      $priority   The priority. The higher this value, the
+     *                             earlier a processor will be executed in
+     *                             the chain (defaults to 0)
+     */
+    public function addMiddleware(callable $middleware, $priority = 0);
```

Comment:
I don't think we should force other implementation to use middlewares...

Reply:
Middleware are a central part of the new client implementation, why shouldn't these methods be in the interface?